File: Java/Conditionals and loops.java
//11 Lesson

//Fill in the blanks to print "Yeah".
int x = 5;
if ( x == 5 ) {
System.out.println("Yeah");
}

//Fill in the blanks to print the greater number.

int x = 10;  int y = 5;

if( x > y ) {

  System.out.println(x);
}
else{

  System.out.println(y);

}

//12 Lesson

//Please fill in the missing parts of the nested if statement to print "it works!" to the screen.
int x = 37;
if (x > 22) {

if(x > 31) {

 System.out.println("it works!");

 }

}

//13 Lesson

//An if statement can contain how many else if statements?
//As many as you want

//14 Lesson

//Fill in the blank to test both conditions in the following if statement.

int age = 23;

int money = 4000;

if (age > 21 && money > 500) {

   System.out.println("Welcome");

}

/*
What is the output of the following code?
int a = 11; int b = 12;
int c = 40;
if (a > 100 || b > 3) {
  System.out.println(a); 
} else {
  System.out.println(c);
}
*/
//11

//15 Lesson

//Fill in the blanks to test the variable's value using the switch statement.

int x = 10;

switch(x) {

  case 10:

    System.out.println("A");

    break;

  
case 20:

    System.out.println("B");

    break; 

}

/*
What is the output of the following code?
int a = 11; int b = 12; int c = 40;
switch (a) {
 case 40:
   System.out.println(b);
   break;
 default:
   System.out.println(c);
}*/

//40

//Fill in the blanks to create a valid switch expression for coffee price.

int price  = switch(coffee) {

  case "Espresso", "Americano" -> 15;

  case "Latte", "Macchiato" -> 20;

  default -> 0;

};

// 16 Lesson

//Rearrange the code to produce a valid finite loop which prints a text to the screen in the loop.

int x = 12;

while (x < 100){
	System.out.println("Java rocks!");
x++;}

/*
How many times will the following loop work?
int x = 0;
int y = 5;
while (x < y) {
  System.out.println("Hello");
  x++;	
}
*/
//5

//17 Lesson

//Drag and drop from the options below to print "Great!" 10 times:

for (int i = 0; i < 10; i++) {

   System.out.println("Great!");


}

//How many times will the following loop run?
//for (int i = 2; i < 10; i = i*i) {
//   System.out.println(i);
//}
//2

//18 Lesson

//How is a do while loop different from a while loop?

//A do while loop runs your code at least one time.

//Fill in the blanks to print the values of the x variable 5 times.


int x = 1;

do {

 System.out.println(x);

 x++;

} 

while(x <= 5);

//19 Lesson

//Fill in the blanks to print "in a loop" 7 times, using the while loop.
int x = 1;

while (x <= 7) {

  System.out.println("in a loop");
  
x++;

}

//Please select the correct statements about && and || operators.

// a || b is true if either a or b is true
//( a || b) && c is true if c is true and either a or b is true

//Fill in the blanks to print "You rock!" if variable "a" is greater than 15, and variable "b" is less than or equal to 72.
int a = 144;

int b = 33;

if (a > 15 &&b <= 72) {

  System.out.println("You rock!");

}

//Fill in the blanks to print "in a loop" 5 times using the for loop.
for
 (int x = 0; x < 5; x++) {

  System.out.println("in a loop");

}


//20 Lesson

/*
Loan Calculator

You take a loan from a friend and need to calculate how much you will owe him after 3 months.
You are going to pay him back 10% of the remaining loan amount each month.
Create a program that takes the loan amount as input, calculates and outputs the remaining amount after 3 months.

Sample Input:
20000

Sample Output:
10628

Here is the monthly payment schedule:
Month 1
Payment: 10% of 20000 = 2000
Remaining amount: 18000
Month 2
Payment: 10% of 18000 = 1800
Remaining amount: 16200
Month 3:
Payment: 10% of 16200 = 1620
Remaining amount: 14580
*/

import java.util.Scanner;

public class Program
{
	public static void main(String[] args) {
		Scanner scanner = new Scanner(System.in);
		int amount = scanner.nextInt();
		int loan = 20000;
		for (int i=0;i<=2;i++){
			loan = loan - loan * 1/10;
		}
		System.out.println(loan);
	}
}

File: C#/Conditionals and Loops.cs
// 13 Lesson

//Fill in the blanks to display Welcome to the screen when age is greater than 16.

int age = 24;

if(age > 16) 

{

   Console.WriteLine("Welcome");
}

//Which is the correct operator for equality testing?

//==

//Fill in the blanks to find the larger of two variables.

int a = 42;

int b = 88;

if(a > b)

{

   Console.WriteLine(a);

} 

else{

   Console.WriteLine(b);

}

/*
What is the output of this code?
int a = 8;
int b = ++a;
if(a > 5)
   b -= 3;
else
   b = 9;
Console.WriteLine(b);
*/

//6

//How many nested if statements can an if statement contain?

//As many as you want


// 14 Lesson

//Fill in the blanks to create a valid switch statement.
int x = 33;

switch(x) {

  case 8:

    Console.WriteLine("Value is 8");

    break;

  case 18:


    Console.WriteLine("Value is 18");

    break;

  
case 33:

    Console.WriteLine("Value is 33");

    break;

}

//Drag and drop from the options below to test the x variable with a switch statement.

switch (x) {

  case 10:

    Console.WriteLine("Ten");

    break;

  case 20:

    Console.WriteLine("Twenty");

    break;

  default:

    Console.WriteLine("No match");.

    break;

}

//What would occur if we forget to include a break statement at the end of case code?

//compile error


// 15 Lesson

//Fill in the blanks to display the value of x to the screen three times.
int x = 42;

int num = 0;

while(num < 3) {

  Console.WriteLine(x);

  num++;

}

//Fill in the blanks to increment the value of num by 2 to display only even values.
int num = 0;

while (num < 100)
{

  Console.WriteLine(num);

  num+=2;

}

//How many times will the following loop execute?

int x = 1;
while (x++ < 5)
{
    if (x % 2 == 0)
        x += 2;
}

//2

// 16 Lesson

//Drag and drop from the options below to create a general for loop:
for ( int x = 5; x < 10; x++;)
{

  Console.WriteLine(x);

}

//Fill in the blanks to print the EVEN values from 0 to 100 using a for loop:
for (int x = 0; x < 100; x+=2)

{
  Console.WriteLine(x);

}

//How many times will this loop run?
int x = 1;
for ( ; x < 7; )
{
  x+=2;
}

//3

// 17 Lesson

//Fill in the blanks to create a valid loop.
int x = 0;

do {

  Console.WriteLine(x);

  x+=2;

} while(x < 10);

/*
What is the output of this code?
int a = 2;
do {
  a+=3;
} while(a < 4);
Console.Write(a);
*/

//5

// 18 Lesson

/*What is the largest number that will be printed by this code?
for (int x = 1; x < 8; x++) {
  if (x > 5)
    break;
  Console.WriteLine(x);
}
*/

//5

//Fill in the blanks to print only even numbers.
for(int x=0; x<99; x++) {

  if(x%2 != 0)

    continue;

  Console.WriteLine(x);

}

// 19 Lesson

//The result of a&&b is true if:

//Both a and b are true

//How many && operators can be used in one if statement?

//As many as you want

/*
What is the output of this code?
int x = 5; int y = 12;
if(x>10 || y/x > 1)
  Console.Write(y-x);
else
  Console.Write(y);
*/

//7

//If a is true and b is false, what is the result of !(a&&b)?

//true

// 20 Lesson

/*
What is the value of x after this code?
int x = 5;
int y = 3;
x = (x > y) ? y : x;
*/

//3

// 21 Lesson

/*
How many times would this loop run?
do { }
while(false);
*/

//one

//Which of the following is used to take user input?

//Console.ReadLine

// 22 Lesson

//Fill in the blanks to print the value of x five times.
int x = 42;

int num = 0;

while(num < 5) {

   Console.WriteLine(x);

   num++; 

}


//Drag and drop from the options below to create a valid finite for loop.

for(int x=0;x<10;x++) {

  Console.WriteLine(x);

}

//Select the correct statements about && and || operators.

//(a && b) || c is true if c is true
// a || b is true if either a or b is true

//Fill in the blanks to calculate the sum of all whole numbers from 1 to 100.
int sum = 0;

for(int x=1;x<=100;x++) {

  sum += x;

}

Console.WriteLine(sum);

/*
What is the value of x after this code?
int x = 4; int y = 9;
x = (y%x != 0) ? y/x : y;
*/

//2


// 23 Lesson

/*
Multiple of 3


You are an elementary school teacher and explaining multiplication to students.
You are going to use multiplication by 3 as your example.
The program you are given takes N number as input. Write a program to output all numbers from 1 to N, replacing all numbers that are multiples of 3 by "*".

Sample Input
7

Sample Output
12*45*7
*/

using System;
using System.Collections.Generic;


namespace SoloLearn
{
    class Program
    {
        static void Main(string[] args)
        {
            int number = Convert.ToInt32(Console.ReadLine());
            
            for(int i=1; i<=number;i++){
                if(i%3==0){
                    Console.Write("*");
                }
                else{
                    Console.Write(Convert.ToString(i));
                }
            }
            
        }
    }
}